src/python/evaluation/qodana/dataset_marking.py

```diff
@@ -0,0 +1,282 @@
+             f"'{ColumnName.CODE.value}' column. The '{ColumnName.LANG.value}' must belong to one of the following "
+             f"values: {', '.join(LanguageVersion.values())}. "
+             f"If '{ColumnName.LANG.value}' is not equal to any of the values, the row will be skipped.",
+    )
+
+    parser.add_argument(
+        "inspections_output_path",
+        type=lambda value: Path(value).absolute(),
+        help="Path where id of all found inspections will be saved.",
+    )
+
+    parser.add_argument("-c", "--config", type=lambda value: Path(value).absolute(), help="Path to qodana.yaml")
+
+    parser.add_argument(
+        "-l",
+        "--limit",
+        type=positive_int,
+        help="Allows you to read only the specified number of first rows from the dataset.",
+    )
+
+    parser.add_argument(
+        "-s",
+        "--chunk-size",
+        type=positive_int,
+        help="The number of files that qodana will process at a time.",
+        default=5000,
+    )
+
+    parser.add_argument(
+        "-o",
+        "--dataset-output-path",
+        type=lambda value: Path(value).absolute(),
+        help="The path where the marked dataset will be saved. "
+             "If not specified, the original dataset will be overwritten.",
+    )
+
+
+@dataclass(init=False)
+class InspectionData:
+    package: str
+
+    def __init__(self, package: str, **kwargs: Any):
+        self.package = package
+
+    def __str__(self):
+        return self.package
+
+
+class DatasetMarker:
+    dataset_path: Path
+    config: Optional[Path]
+    limit: Optional[int]
+    chunk_size: Optional[int]
+    inspection_to_id: Dict[str, int]
+    dataset_output_path: Path
+    inspections_output_path: Path
+
+    def __init__(self, args: Namespace):
+        self.dataset_path = args.dataset_path
+        self.config = args.config
+        self.limit = args.limit
+        self.chunk_size = args.chunk_size
+
+        self.dataset_output_path = self.dataset_path
+        if args.dataset_output_path is not None:
+            self.dataset_output_path = args.dataset_output_path
+
+        self.inspections_output_path = args.inspections_output_path
+
+        self.inspection_to_id = {}
+
+    def mark(self):
+        df = pd.read_csv(self.dataset_path, index_col=ColumnName.ID.value, nrows=self.limit)
+
+        group_by_lang = df.groupby(ColumnName.LANG.value)
+        unique_languages = df[ColumnName.LANG.value].unique()
+
+        logger.info(f"Unique languages: {unique_languages}")
+
+        groups = []
+        for language in unique_languages:
+            lang_group = group_by_lang.get_group(language)
+
+            if language in LanguageVersion.values():
+                try:
+                    logger.info(f"Processing the language: {language}")
+                    groups.append(self._mark_language(lang_group, LanguageVersion(language)))
+                except NotImplementedError:
```

Comment:
I think we should leave the handling of this error. Because if the dataset contains a language that is in LanguageVersions, but is not supported by this script, the dataset processing will just crash.